contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -341,7 +348,9 @@ func (k *kubeScheduler) doSchedule(task *podtask.T, err error) (string, error) {
 		}
 
 		task.Offer = offer
-		k.api.algorithm().Procurement()(task, details) // TODO(jdef) why is nothing checking the error returned here?
+		if err := k.api.algorithm().Procurement()(task, details); err != nil {
+			return "", err
```

Comment:
forgot to mention earlier, we probably want to `offer.Release()` before returning